src/sth_database.js

```diff
@@ -5,7 +5,109 @@
 
   var mongoose = require('mongoose');
 
-  var sthConfig, sthLogger, sthHelper, connectionURL, eventSchema, aggregatedSchema;
+  var sthConfig, sthLogger, sthHelper, connectionURL, connection, eventSchema, aggregatedSchema;
+
+  /**
+   * Declares the Mongoose schemas.
+   */
+  function defineSchemas() {
+    switch (sthConfig.DATA_MODEL) {
+      case sthConfig.DATA_MODELS.COLLECTIONS_PER_SERVICE_PATH:
+        eventSchema = mongoose.Schema({
+          timestamp: Date,
+          entityId: String,
+          entityType: String,
+          attributeName: String,
+          attributeType: String,
+          attributeValue: Number
+        });
+        break;
+      case sthConfig.DATA_MODELS.COLLECTIONS_PER_ENTITY:
+        eventSchema = mongoose.Schema({
+          timestamp: Date,
+          attributeName: String,
+          attributeType: String,
+          attributeValue: Number
+        });
+        break;
+      case sthConfig.DATA_MODELS.COLLECTIONS_PER_ATTRIBUTE:
+        eventSchema = mongoose.Schema({
+          timestamp: Date,
+          attributeType: String,
```

Comment:
Yes, that's correct: the attribute type of a given attribute of a given entity could be changed at Orion using its API.